doc/sphinx/proof-engine/tactics.rst

```diff
@@ -898,314 +1113,235 @@ Applying theorems
-   then the tactic implements a mix of the "Let" and "Conv".
+      There is nothing to introduce even after :tacn:`hnf` has been completely applied.
 
-   If the current goal is a dependent product :g:`forall x:T, U`
-   (resp :g:`let x:=t in U`) then :tacn:`intro` puts :g:`x:T` (resp :g:`x:=t`)
-   in the local context. The new subgoal is :g:`U`.
+   .. _intro_examples:
+   .. example:: `intro` and `intros`
 
-   If the goal is a non-dependent product :math:`T \rightarrow U`, then it
-   puts in the local context either :g:`Hn:T` (if :g:`T` is of type :g:`Set`
-   or :g:`Prop`) or :g:`Xn:T` (if the type of :g:`T` is :g:`Type`).
-   The optional index ``n`` is such that ``Hn`` or ``Xn`` is a fresh
-   identifier. In both cases, the new subgoal is :g:`U`.
+      .. coqtop:: reset out
 
-   If the goal is an existential variable, :tacn:`intro` forces the resolution
-   of the existential variable into a dependent product :math:`\forall`\ :g:`x:?X, ?Y`,
-   puts :g:`x:?X` in the local context and leaves :g:`?Y` as a new subgoal
-   allowed to depend on :g:`x`.
+         Goal forall m n, m < n -> (let x := 0 in True).
 
-   The tactic :tacn:`intro` applies the tactic :tacn:`hnf`
-   until :tacn:`intro` can be applied or the goal is not head-reducible.
+      .. coqtop:: all
 
-   .. exn:: No product even after head-reduction.
-      :undocumented:
+         intro m.
+         intro n.
+         intro H.
+         intro x.
 
-   .. tacv:: intro @ident
+      This single `intros` tactic is equivalent to the 4 preceding `intro` tactics:
 
-      This applies :tacn:`intro` but forces :token:`ident` to be the name of
-      the introduced hypothesis.
+      .. coqtop:: reset out
 
-      .. exn:: @ident is already used.
-         :undocumented:
+         Goal forall m n, m < n -> (let x := 0 in True).
 
-   .. note::
+      .. coqtop:: all
 
-      If a name used by intro hides the base name of a global constant then
-      the latter can still be referred to by a qualified name
-      (see :ref:`Qualified-names`).
+         intros m n H x.
 
-   .. tacv:: intros
-      :name: intros
+.. tacn:: intros {* @intropattern }
+          intros until {| @ident | @natural }
 
-      This repeats :tacn:`intro` until it meets the head-constant. It never
-      reduces head-constants and it never fails.
+      The first form introduces zero or more items into the context from the
+      constructs listed in :tacn:`intro`.  If :n:`@intropattern` is not specified,
+      the tactic introduces items until it reaches the :term:`head constant`.
 
-   .. tacv:: intros {+ @ident}.
+      If :n:`@intropattern` is specified, the :tacn:`hnf` tactic is applied until
+      it finds an item that can be introduced into the context.
+      The :n:`@intropattern` is
+      often just a list of :n:`ident`\s, but other forms can also be specified
+      in order to, for example, introduce all :n:`forall` :term:`variables <variable>` (`*`);
```

Comment:
This definition does not seem to be appropriate in this context (it talks about variables already introduced in the context). I suggest to just remove this link (forall variables is clear enough, and is even used in the linked definition BTW).
```suggestion
      in order to, for example, introduce all :n:`forall` variables (`*`);
```